### PR TITLE
Stable solution for Twosum Problem with duplicates

### PR DESCRIPTION
**Pull request by prabu-manickam**
My contribution towards the TwoSum problem
Data Structure : HashMap
Time Complexity : O(n)
Space Complexity : O(n)

### CODE REVIEW

File: algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                }
+            }
+            
+            if(pairPos != null) {
+                return new int[]{i, (int)pairPos};
+            }
+        }
+        return null;
+    }
+
```
Comment (prashantchawke) [inline algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java]: I Liked your solution. It's way better in execution time.
I did little improvements. Just to make decisions while iterating over array itself.
It reduces execution time to half in leetcode IDE.
```java
public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> elementIndexMap = new HashMap<>();
        for (int i = 0; i <= nums.length - 1; i++) {
            //If ELEMENT index exists
            if (elementIndexMap.get(nums[i]) == null) {
                // If COMPLEMENT OF ELEMENT's index exists
                if ((elementIndexMap.get(target - nums[i]) != null)) {
                    return new int[]{elementIndexMap.get(target - nums[i]), i};
                }
                elementIndexMap.put(nums[i], i);
            } // if element's index exists and element is half of target
            else if (2 * nums[i] == target) {
                return new int[]{elementIndexMap.get(nums[i]), i};
            }
        }
        return null;
    }
 ```

Comment (prabu-manickam) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java]: Wow.. It is a very simplified solution! I liked it..

Comment (prabu-manickam) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java]: Can make the changes in the solution. lets also list the solutions in the  ` README.md ` ?

Comment (prabu-manickam) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java]: shall we find the best solution in term of execution in leetcode if possible?

Comment (prabu-manickam) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java]: I have further simplified the solution as follows, what do you think?

`
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pos = new HashMap<>();
        for(int i = 0; i<nums.length; i++) {
            if( pos.get(target-nums[i]) != null) {
              return new int[]{pos.get(target-nums[i]), i};  
            } else {
              pos.put(nums[i], i);
            }
        }
     return null;
    }
}
`

Comment (prabu-manickam) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java]: Updated the code, check once?

File: algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+*/
+
+class TwoSumWithDuplicatesStableSolution {
+    public int[] twoSum(int[] nums, int target) {
+        Map<Integer, Integer> pos = new HashMap<>();
+        
+        for(int i = 0; i<nums.length; i++) {
+            if( pos.get(target-nums[i]) != null) {
+              return new int[]{pos.get(target-nums[i]), i};  
+            } else {
```
Comment (prashantchawke) [inline algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java:18]: Nice. Just this else is not required.

Comment (prashantchawke) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java:18]: :+1:

File: algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java (hunk 1 above)
Comment (prashantchawke) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java]: Yes We should get teh best solution every tiem and then merge it :+1:

File: algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java (hunk 2 above)
Comment (prabu-manickam) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java:18]: you mean to remove the else part here?

Comment (prashantchawke) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java:18]: Just a silly thing.
```java
        for(int i = 0; i<nums.length; i++) {
            if( pos.get(target-nums[i]) != null) {
              return new int[]{pos.get(target-nums[i]), i};  
            }
              pos.put(nums[i], i);
        }
```

Comment (prashantchawke) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java:18]: Now. It's exactly same as the best answer provided in the leetcode now. :+1:

Comment (prabu-manickam) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java:18]: Wow! thats great!~ Shall we update the solution and list in the readme?

Comment (prashantchawke) [reply algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java:18]: Sure. Prabu!. Let's make this weekend a golden one.

**Pull request merged**